### PR TITLE
Add configuration for marking stale PRs

### PR DESCRIPTION
**Pull request by axic**
Add Github Actions configuration and uses https://github.com/actions/stale.

### CODE REVIEW

File: action.yml
Comment (axic) [inline action.yml:13]: Any suggestion what to put here?

File: action.yml
Comment (axic) [inline action.yml:14]: And here?

File: action.yml
Comment (axic) [inline action.yml:16]: These two are the default settings but added them here for explicitness.

File: action.yml
Comment (axic) [inline action.yml:17]: The default is "Stale" but we are using lowercase labels.

**Comment by axic** [comment]
@MicahZoltu @lightclient @Souptacular I have no idea how this works and how to test it, apart from merging it...

File: .github/workflows/greetings.yml
Comment (axic) [inline .github/workflows/greetings.yml:13]: Not sure if this a good way to raise awareness more, but the issue/PR templates I think are mostly ignored by people.

File: .github/workflows/stale.yml
Comment (MicahZoltu) [inline .github/workflows/stale.yml:14]: We should probably let them know *how* to reach out to an editor.  For me, just commenting on the pull request is enough to get my attention.  I can't speak for the process of the other editors but if I was the only editor I would want the messaging to go something like:
> There has been no activity on this pull request for two months.  It will be closed in a week if no further activity occurs.  If you would like to move this EIP forward, please respond to any outstanding feedback or add a comment indicating that you have addressed all *required* feedback and are ready for a review.

File: .github/workflows/stale.yml
Comment (MicahZoltu) [inline .github/workflows/stale.yml:15]:
```suggestion
         close-pr-message: 'This pull request was closed due to inactivity. If you are still pursuing it, feel free to reopen it and respond to any feedback or request a review in a comment.'
```

**Review by MicahZoltu — APPROVED** [review]
I left some feedback, but I'm good to merge this as-is or with the changes I suggested.  As with most things, I prefer to just give it a try and roll it back if things don't work out.  I think the worst case failure here would be that we got the configuration wrong and it instantly closes everything that hasn't been touched for 7 days.  In that case, someone can just go through and re-open them all with some stock apology for the confusion.  May take 15 minutes of work to clean up the mess.

File: .github/workflows/stale.yml
Comment (axic) [reply .github/workflows/stale.yml:14]:
```suggestion
        stale-pr-message: 'There has been no activity on this pull request for two months. It will be closed in a week if no further activity occurs. If you would like to move this EIP forward, please respond to any outstanding feedback or add a comment indicating that you have addressed all required feedback and are ready for a review.'
```

Comment (axic) [reply .github/workflows/stale.yml]: Actually the new wording suggests a pull request can only be a new EIP, but there are plenty of modifications pending author's approval. Not sure how much lengths we have to go to get this right though 😕

**Review by lightclient — APPROVED** [review]
If we want to give it a dry run, I believe we can pass `debug-only: true` as an argument to the actions (just going off the actions README). With that said, I would also be happy to go through and help reopen PRs if everything older than 7 days is closed. Given that this is in the `actions` organization, has recent commits, and has no major open issues in the tracker, I think merging is reasonably safe.

**Comment by axic** [comment]
Okay, let's roll... 😬

**Comment by axic** [comment]
@lightclient the cron job does not seem to be picked up, even though this configuration came from the "github actions wizard". Do you have any idea / any experience with github actions?

**Comment by lightclient** [comment]
@axic I'm not the best with cron, but isn't `0 0 * * *` supposed to trigger everyday at 12am? I'm not sure what timezone GitHub is running on. Since this was only merged 5 hours ago, that wouldn't have triggered on PT or UTC yet.

**Comment by axic** [comment]
The manual I looked at said `<seconds> <minutes> <hours> <days of month> <months> <days of week> <years>`, so it would be every top of the hour.

**Comment by axic** [comment]
Hm, maybe there are various versions of cron formats out there. That's a good hint, will check.

**Comment by axic** [comment]
From https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onschedule
> You can schedule a workflow to run at specific UTC times using POSIX cron syntax.

Which brings to https://pubs.opengroup.org/onlinepubs/9699919799/utilities/crontab.html#tag_20_25_07:
> In the POSIX locale, the user or application shall ensure that a crontab entry is a text file consisting of lines of six fields each. The fields shall be separated by <blank> characters. The first five fields shall be integer patterns that specify the following:
> 
> 1. Minute [0,59]
> 2. Hour [0,23]
> 3. Day of the month [1,31]
> 4. Month of the year [1,12]
> 5.Day of the week ([0,6] with 0=Sunday)

Will fix it 😉

**Comment by axic** [comment]
Screenshot

Not sure why it still is not working.

**Comment by lightclient** [comment]
Hmm, my only guess is that maybe cron jobs aren't added immediately once they're merged to master. It's possible there is a cron job which picks up new cron jobs. Alternatively, maybe putzing around the GitHub actions tab could yield some information?

**Comment by axic** [comment]
You may be right, at least something picked it up and it ran at 20 past :)

**Comment by axic** [comment]
Now I'm thinking that maybe we should use different values than the default 2 month + 1 week:
- 1 month to be marked stale - this should give enough time to respond to incoming new PRs
- 2 weeks to be closed - I think 1 week might be short for people to respond to the bot and they will end up opening new PRs which are then abandoned, starting a cycle

**Pull request merged**
